### ISSUE TITLE
Obtaining ligand names from Transformations in failed Tasks

### ISSUE THREAD

**Issue by ijpulidos** [OpenFreeEnergy/gufe#243]
From https://github.com/choderalab/asapdiscovery/pull/477 we would find convenient if there's a way to extract the ligand names from a `Transformation` (or ProtocolDAGResult`?) in a failed `Task`.  This is convenient when debugging why a specific task/transformation has failed.

Right now, we would have to manually inspect the `Transformation` object and then inspect its states (which are `ChemicalSystem`s `stateA` and `stateB`) and then inspect the `Component`s in each of them to extract the ligands and their names. I raise this issue to discuss the possibility of having a helper function/method that can be used directly in a transformation (or maybe even a `ProtocolDAGResult`) that can help obtaining the ligand names involved in the transformation.

We would need to think about the complexities when the transforming components are not ligands (`SmallMoleculeComponent`) but protein/receptors or solvent.

**Comment by ijpulidos** [comment]
I think what is pretended in #144 can help solving this issue

**Comment by dwhswenson** [comment]
I agree that something like this would be useful, and somewhat depends on the ideas in #144. FWIW, we've avoided the problem by giving explicit names to the systems and transformations, which serves the same purpose for us. Examples transformation names:

* easy_rbfe_lig_ejm_31_complex_lig_ejm_42_complex
* easy_rbfe_lig_ejm_31_solvent_lig_ejm_42_solvent

The "easy_rbfe" is a prefix that @RiesBen added; then you have the two states names, which include the ligand name and whether it is complex or solvent.

This naming is automatic if you make the `AlchemicalNetwork` from a `LigandNetwork` using the `ligand_network.to_rbfe_alchemical_network` method.

(I agree that a more robust solution would be good, but this is an easy workaround to solve the problem today.)
